packages/multichain/src/scope/validation.test.ts

```diff
@@ -0,0 +1,185 @@
+      ],
+      [
+        true,
+        'the scopeString is a valid CAIP chainId and the scopeObject is valid',
+        'eip155:1',
+        validScopeObject,
+      ],
+      [
+        false,
+        'the scopeString is a CAIP chainId but references is nonempty',
+        'eip155:1',
+        {
+          ...validScopeObject,
+          references: ['5'],
+        },
+      ],
+      [
+        false,
+        'the scopeString is a valid CAIP namespace but references are invalid CAIP references',
+        'eip155',
+        {
+          ...validScopeObject,
+          references: ['@'],
+        },
+      ],
+      [
+        false,
+        'methods contains empty string',
+        validScopeString,
+        {
+          ...validScopeObject,
+          methods: [''],
+        },
+      ],
+      [
+        false,
+        'methods contains non-string',
+        validScopeString,
+        {
+          ...validScopeObject,
+          methods: [{ foo: 'bar' }],
+        },
+      ],
+      [
+        true,
+        'methods contains only strings',
+        validScopeString,
+        {
+          ...validScopeObject,
+          methods: ['method1', 'method2'],
+        },
+      ],
+      [
+        false,
+        'notifications contains empty string',
+        validScopeString,
+        {
+          ...validScopeObject,
+          notifications: [''],
+        },
+      ],
+      [
+        false,
+        'notifications contains non-string',
+        validScopeString,
+        {
+          ...validScopeObject,
+          notifications: [{ foo: 'bar' }],
+        },
+      ],
+      [
+        false,
+        'notifications contains non-string',
+        'eip155:1',
+        {
+          ...validScopeObject,
+          notifications: [{ foo: 'bar' }],
+        },
+      ],
+      [
+        false,
+        'unexpected properties are defined',
+        validScopeString,
+        {
+          ...validScopeObject,
+          unexpectedParam: 'foobar',
+        },
+      ],
+      [
+        true,
+        'only expected properties are defined',
+        validScopeString,
+        {
+          references: [],
+          methods: [],
+          notifications: [],
+          accounts: [],
+          rpcDocuments: [],
+          rpcEndpoints: [],
+        },
+      ],
+    ])(
+      'returns %s when %s',
+      (
+        expected: boolean,
+        _scenario: string,
+        scopeString: string,
+        scopeObject: unknown,
+      ) => {
+        expect(
+          isValidScope(scopeString, scopeObject as ExternalScopeObject),
+        ).toStrictEqual(expected);
+      },
+    );
+  });
+
+  describe('validateScopes', () => {
+    const validScopeObjectWithAccounts = {
+      ...validScopeObject,
+      accounts: [],
+    };
+
+    it('does not throw an error if required scopes are defined but none are valid', () => {
+      expect(
+        validateScopes(
+          { 'eip155:1': {} as unknown as ExternalScopeObject },
```

Comment:
If we know that this is not an `ExternalScopeObject` and so we expect this to fail type checking, should we indicate that instead?

``` suggestion
          // @ts-expect-error Intentionally invalid input
          { 'eip155:1': {} },
```

(similar for below cases)

Reply:
https://github.com/MetaMask/core/pull/4784/commits/02f35f599bd8a1199fb6782aba30c50784541cf8